### PR TITLE
Update doc strings to reflect accurate perm option

### LINKED ISSUE #8784: API reference for SasPermission objects list incorrect operations.
For example, the below is for BlobSasPermissions and lists "read, write, delete, list", but the correct permission options for BlobSasPermissions is "read, create, write, add, delete".

```
permission
str or BlobSasPermissions
The permissions associated with the shared access signature. The user is restricted to operations allowed by the permissions. Permissions must be ordered read, write, delete, list. Required unless an id is given referencing a stored access policy which contains this field. This field must be omitted if it has been specified in an associated stored access policy.

default value: None
```

Scrub all doc strings to ensure they list the correct permission options per SasPermission object.

### PR DESCRIPTION
**Pull request by jongio**
Resolves #8784 

Note that the datalake tests are failing with tox so I wasn't able to fullly test, but given that these are only doc strings we should be ok.  See issue: #8789

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def generate_directory_sas(
     :param permission:
         The permissions associated with the shared access signature. The
         user is restricted to operations allowed by the permissions.
-        Permissions must be ordered read, write, delete, list.
+        Permissions must be ordered read, add, create, write, delete.
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py]: Thanks so much for taking time to find the proper permission we want to expose to user! ^_^
let's make it read, create, write, delete(add is for append blob, in datalake all files are acatually a blockblob)

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py [hunk 2]
```diff
@@ -316,7 +316,7 @@ def generate_file_sas(
...
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py]: same here ^_^ lets make it `read, write, delete, list`

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py [hunk 3]
```diff
@@ -137,7 +137,7 @@ def generate_share(self, share_name, permission=None, expiry=None,
         :param ShareSasPermissions permission:
             The permissions associated with the shared access signature. The
             user is restricted to operations allowed by the permissions.
-            Permissions must be ordered read, create, write, delete, list.
+            Permissions must be ordered read, write, delete, list.
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py]: Is there a reason we want to delete create here? https://docs.microsoft.com/en-us/rest/api/storageservices/create-service-sas#permission-for-a-share

Comment (jongio) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py]: I based it off of the class: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py#L725

Which has read, write, delete, list.

Are you saying the class is incorrect?

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py (hunk 1 above)
Comment (jongio) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py]: It is typed as BlobSasPermissions, which has read, add, created, write, delete
https://github.com/Azure/azure-sdk-for-python/blob/6f606828aeb49e1259d8d7af5da05a559bb84237/sdk/storage/azure-storage-blob/azure/storage/blob/_models.py#L959

We may need to subclass to account for directory.

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py (hunk 3 above)
Comment (jongio) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py]: I based it off of the class: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py#L725

Which has read, write, delete, list.

Are you saying the class is incorrect?

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py (hunk 2 above)
Comment (jongio) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py]: This is typed as BlobSasPermissions, see comment above.

**Comment by jongio** [comment]
@chlowell may want to review due to being related to auth

**Comment by xiafu-msft** [comment]
Sorry for the late reply, I'm looking at the comments now

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py (hunk 1 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py]: I found the doc for this file is not very accurate! Thanks for helping me finding this. https://github.com/Azure/azure-sdk-for-python/blob/5f3b14d7260831fdc1404600192fb759e49bc85f/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py#L21-L24 I created a [PBI](https://msazure.visualstudio.com/One/_workitems/edit/5815390) to fix the doc. 

Here is the [DirectorySasPermission ](https://github.com/Azure/azure-sdk-for-python/blob/5f3b14d7260831fdc1404600192fb759e49bc85f/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py#L403) class
https://github.com/Azure/azure-sdk-for-python/blob/5f3b14d7260831fdc1404600192fb759e49bc85f/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py#L403-L422

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py (hunk 2 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py]: replied above ^_^

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py (hunk 3 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py]: I think the class is not accurate.. The rest doc I pasted there should be accurate

I created a issue(Rakshith linked the issue in the conversation) which is to make every class accurate according to rest doc

Comment (jongio) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py]: Can you link to that issue here?  Thanks

**Comment by xiafu-msft** [comment]
https://github.com/Azure/azure-sdk-for-python/issues/8799

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py (hunk 3 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py]: Done 😊

Comment (lmazuel) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared_access_signature.py]: For clarity, pasting the link actually in this conversation:
https://github.com/Azure/azure-sdk-for-python/issues/8799

**Comment by lmazuel** [comment]
Discussed with @xiafu-msft that she will starts a new PR from scratch to fix #8784, instead of pushing additional commit here.

**Pull request closed without merging**